extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisReceiver.scala

```diff
@@ -216,6 +226,25 @@ private[kinesis] class KinesisReceiver[T](
     shardIdToLatestStoredSeqNum.get(shardId)
   }
 
+  /** Pass the setCheckpointer request to the checkpointer. */
+  private[kinesis] def setCheckpointerForShardId(
```

Comment:
Since I have the `getCheckpointer` method, I didn't want them to feel the same way. `getCheckpointer` returns the central `checkpointer` but `setCheckpointer` is for a specific shardId. Don't you think that would be confusing?